src/main/java/io/github/bookcrawler/EmpikBookExtractor.java

```diff
@@ -0,0 +1,60 @@
+package io.github.bookcrawler;
+
+import io.github.bookcrawler.entities.BookInfo;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class EmpikBookExtractor implements BookExtractor {
+
+    public Iterable<BookInfo> extract() throws IOException {
+        Connection connect = Jsoup.connect("http://www.empik.com/ebooki/promocje/");
+        Set<BookInfo> books = new HashSet<BookInfo>();
+
+
+        Document document = connect.get();
+        Elements contentPacket = document.getElementsByAttributeValue("class", "more_lnk");
+
+        List<String> sectionUrls = new ArrayList<String>();
+        for (Element element : contentPacket) {
+            sectionUrls.add("http://www.empik.com" + element.attr("href"));
+        }
+
+        for (String url : sectionUrls) {
+            books.addAll(getBooksFromSection(connect.url(url).get()));
+        }
```

Comment:
27 to here it's O(n^2). We first iterate through collection to fill other collection, then we iterate that other collection.

Consider stream:
```
        contentPacket.stream().map(e -> EMPIK + href(e)).forEach(s -> {
            try {
                books.addAll(getBooksFromSection(connect.url(s).get()));
            } catch (IOException e) {
                // TODO: logging here
            }
        });
        return books;
```
It flows better and has the option of being parallelized (`parallelStream`).